webapp/src/DataTransferObject/Shadowing/ProblemEvent.php

```diff
@@ -7,7 +7,7 @@ class ProblemEvent implements EventData
     public function __construct(
         public readonly string $id,
         public readonly string $name,
-        public readonly int $timeLimit,
+        public readonly ?int $timeLimit,
```

Comment:
I don't think we should merge into `main` things that are basically workarounds for bugs in PC^2's implementation. I'd rather we keep that in a separate branch and make sure PC^2 fixes this. Otherwise, we'll not notice anymore when such bugs occur in other systems we shadow and I think it's better to be strict here, also because the alternative leads to code complexity creep.

Reply:
Is time limit a required property? If so I agre

Reply:
> Is time limit a required property? If so I agree

It seems to be, I'll revert.

Reply:
According to https://ccs-specs.icpc.io/2023-06/contest_api#problems it is.

Reply:
Also depends on whether all users have access to it. I will check the current access policy when I’m at my pc to see if there is any role that doesn’t have access

Reply:
The contest information access policy is not explicit on this: it says teams can't access 'other properties' from problems. But I added a comment there to discuss.

Reply:
Are the time limits not visible in the problem set, or at least on the problems page view for the teams? I'd strongly push to make this public info, at least in WF/ICPC setting.

Reply:
That is why I added the comment, I'd argue the same.